bee-ternary/src/trit/mod.rs

```diff
@@ -11,23 +11,36 @@
 
 use std::{convert::TryFrom, fmt, hash};
 
+/// Balanced trits.
 pub mod balanced;
+/// Unbalanced trits.
 pub mod unbalanced;
 
 // Reexports
 pub use self::{balanced::Btrit, unbalanced::Utrit};
 
+/// A trait implemented by both balanced ([`Btrit`]) and unbalanced ([`Utrit`]) trits.
 pub trait Trit:
     Copy + Sized + fmt::Debug + hash::Hash + Into<i8> + Ord + PartialEq + ShiftTernary + TryFrom<i8> + fmt::Display
 {
+    /// Attempt to increment the value of this trit, returning [`None`] if an overflow occurred.
     fn checked_increment(self) -> Option<Self>;
+
+    /// The zero value of this trit.
     fn zero() -> Self;
 
+    /// Turn this trit reference into one with an arbitrary lifetime.
+    ///
+    /// Note that this is largely an implementation detail and is rarely useful for API users.
     fn as_arbitrary_ref<'a>(&self) -> &'a Self;
 }
 
+/// A trait implemented by trits that can be shifted between balance domains.
+// TODO: Is this a good API?
```

Comment:
What's the issue with it ?

Reply:
It feels bizarre for the same method to shift both ways. `.into_balance()` / `.into_unbalanced()` seems like the more natural API to me.

Reply:
But then you have to know the concrete type ? You can't "toggle" it anymore ?

Reply:
Sure, I'm just not convinced that's always what is needed.